### PR TITLE
connection-status-feature

### LINKED ISSUE #17008: Angular
- [ ] Create a service store internet connection status.(use InternalStore<> )
- [ ] create a component and add app.component.ts (on template and dev app. it must be sibling with <abp-loader-bar />)
- [ ] Work with design team ( @armgnunlu )

### PR DESCRIPTION
**Pull request by Sinan997**
Resolves https://github.com/abpframework/abp/issues/17008

This PR adding a feature that shows an icon at top right corner when user internet connection is goes down

the icon is blinking between #c1c1c1 and #DC3545 color

### How to test it?

you need to disconnect the internet to test it, you can do that via developers tool;
f12 + application + service workes + offline

### CODE REVIEW

File: npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @media (width < 768px){
+      .status-icon{
+        top: 80px;
+      }
+    }
+  `]
+})
+export class InternetConnectionStatusComponent implements OnInit{
+  internetConnection$ = of(true);
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts]: Handle state that unkown of internet status

File: npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  internetConnectionService = inject(InternetConnectionService)
+
+  ngOnInit(): void {
+    this.internetConnection$ = this.internetConnectionService.getStatus$()
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts]: You don't need to call  at onInit. you should do it in constructor level. 
  internetConnectionService = inject(InternetConnectionService)
   internetConnection$ = this.internetConnectionService.getStatus$()

File: npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { fromEvent, merge, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface InternetConnectionState{
```
Comment (rqx110) [inline npm/ng-packs/packages/core/src/lib/services/internet-connection-service.ts]: no need this interface any more

File: npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Component, inject } from '@angular/core';
+import { CommonModule } from '@angular/common'
+import { map } from 'rxjs';
+import { InternetConnectionService, LocalizationModule } from '@abp/ng.core';
+
+@Component({
+  selector: 'abp-internet-status',
+  standalone: true,
+  imports:[CommonModule,LocalizationModule],
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/internet-connection-status/internet-connection-status.component.ts:9]: use ngIfModule insead of common

**Review by mahmut-gundogdu — APPROVED** [review]

**Pull request merged**